Makefile

```diff
@@ -16,70 +9,52 @@ all: docs packages
 test: tmp
-	python3 -m pytest -vvv -ra --showlocals tests/unit
+	hatch run test:unit
 smoke: tmp
-	python3 -m pytest -vvv -ra --showlocals tests/unit/test_cli.py
-coverage: tmp
-	coverage run --source=tmt,bin -m pytest -vvv -ra --showlocals tests
-	coverage report
-	coverage annotate
+	hatch run test:smoke
+coverage: tmp nitrateconf
+	hatch run test:cov
+nitrateconf:
+	test -e ~/.nitrate || echo -en '[nitrate]\nurl = https://nitrate.server/xmlrpc/\n' | tee ~/.nitrate
+
 # Regenerate test data for integration tests
 # remove selected/all response files in tests/integration/test_data directory
 requre:
-	cd tests/integration; python3 -m pytest -vvv -ra --showlocals
-	# response files cleanup
-	requre-patch purge --replaces :milestone_url:str:SomeText --replaces :latency:float:0 tests/integration/test_data/test_nitrate/*
-
+	hatch run test:requre
 
 # Build documentation, prepare man page
-docs: man
-	cd docs && make html
-man: source
-	cp docs/header.txt $(TMP)/man.rst
-	tail -n+8 README.rst >> $(TMP)/man.rst
-	# TODO rst2man cannot process this directive, removed for now
-	sed '/versionadded::/d' -i $(TMP)/man.rst
-	rst2man $(TMP)/man.rst > $(TMP)/$(PACKAGE)/tmt.1
+docs: clean
+	hatch run docs:create
+man:
+	hatch run docs:man
 
-
-# RPM packaging and Packit
-source: clean tmp
+# Packaging and Packit
+build: clean man
+	hatch build
+tarball: clean tmp build
 	mkdir -p $(TMP)/SOURCES
-	mkdir -p $(TMP)/$(PACKAGE)
-	cp -a $(FILES) $(TMP)/$(PACKAGE)
-	sed -i $(REPLACE_VERSION) $(TMP)/$(PACKAGE)/tmt/__init__.py
-tarball: source man
-	cd $(TMP) && tar cfz SOURCES/$(PACKAGE).tar.gz $(PACKAGE)
-	@echo ./tmp/SOURCES/$(PACKAGE).tar.gz
-version:
-	@echo "$(VERSION)"
-rpm: tarball
-	rpmbuild --define '_topdir $(TMP)' -bb tmt.spec
-srpm: tarball
+	cp dist/tmt-*.tar.gz $(TMP)/SOURCES
+rpm: tarball ver2spec
+	# If the build system is missing the required dependencies, use nosrc.rpm to install them
+	rpmbuild --define '_topdir $(TMP)' -bb tmt.spec || sudo dnf builddep -y $(TMP)/SRPMS/tmt-*nosrc.rpm && rpmbuild --define '_topdir $(TMP)' -bb tmt.spec
+srpm: tarball ver2spec
 	rpmbuild --define '_topdir $(TMP)' -bs tmt.spec
 packages: rpm srpm
-bump2dev:
-	$(shell sed "s/^Version:.*/Version: $$(echo $(VERSION) | awk '{ split($$0,v,"."); printf "%s.%s.dev", v[1], v[2]+1 }')/" -i tmt.spec)
-
+version:
+	hatch version
+ver2spec:
+	$(shell sed -E "s/^(Version:[[:space:]]*).*/\1$$(hatch version)/" -i tmt.spec)
 
 # Containers
 images:
-	podman build -t tmt --squash -f ./containers/Dockerfile.mini .
-	podman build -t tmt-all --squash -f ./containers/Dockerfile.full .
-
-
-# Python packaging
-wheel:
-	cp -a tmt/__init__.py tmt/__init__.py.backup
-	sed -i $(REPLACE_VERSION) tmt/__init__.py
-	python3 setup.py bdist_wheel
-	mv tmt/__init__.py.backup tmt/__init__.py
-upload:
-	twine upload dist/*.whl
+	podman build -t tmt --squash -f ./containers/Containerfile.mini .
+	podman build -t tmt-all --squash -f ./containers/Containerfile.full .
 
+# Development
+devrpms:
```

Comment:
What about `deps` or `develop`?

Reply:
`deps` is out if we use it for rpmbuild dependencies as suggested above. `make develop` feels odd to me, but am happy to change it.

Reply:
Ok, let's keep `deps` for the missing dependencies and use `develop` here, the name is consistent with the [Develop](https://tmt.readthedocs.io/en/latest/contribute.html#develop) section of the contribution guide.